nemo/collections/asr/models/label_models.py

```diff
@@ -165,8 +180,48 @@ def validation_epoch_end(self, outputs):
 
         return {'val_loss': val_loss_mean, 'log': tensorboard_logs}
 
-    def training_epoch_end(self, outputs):
-        train_acc = (sum([x['n_correct_pred'] for x in outputs]) / sum(x['n_pred'] for x in outputs)) * 100
-        logging.info("training accuracy {:.3f}".format(train_acc))
+    def test_step(self, batch, batch_ix):
+        self.eval()
+        audio_signal, audio_signal_len, labels, _ = batch
+        _, embs = self.forward(input_signal=audio_signal, input_signal_length=audio_signal_len)
+        return {'embs': embs, 'labels': labels}
+
+
+class ExtractSpeakerEmbeddingsModel(EncDecSpeakerLabelModel):
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None, root_dir='.'):
```

Comment:
model's constructor should only take cfg and (optionally) trainer. Please remove root_dir from constructor args
Can't you just take root_dir from cfg?

Reply:
I could, but anyhow considering the version formatting from PTL and hydra, I would have to anyhow overwrite them, so passing this as an arg.

Reply:
@nithinraok Dont have a different signature than the rest of the codebase. Do whatever override of the config you need internally, but signature of constructor can accept only cfg and trainer.